libs/blocks/merch-card/merch-card.js

```diff
@@ -1,184 +1,206 @@
 
   innerElements.forEach((element) => {
-    if (element.tagName.match(/^H[1-6]$/)) element.classList.add(`consonant-${cardType}-title`);
-    if (element.tagName.match(/^P$/)) element.classList.add(`consonant-${cardType}-description`);
-    if (element.tagName.match(/^UL$/)) {
-      list.classList.add(`consonant-${cardType}-list`);
-      list.querySelectorAll('li').forEach((li) => li.classList.add(`consonant-${cardType}-list-item`));
+    const { tagName } = element;
+    if (isHeadingTag(tagName)) {
+      createAndAppendTag(tagName, { slot: textStyles[tagName] }, element.innerHTML, merchCard);
+      return;
+    }
+    if (isParagraphTag(tagName)) {
+      bodySlot.append(element);
     }
   });
-
-  inner.append(...innerElements);
-  addFooter(links, inner, cardType !== PLANS_CARD);
-  decorateFooter(el, altCta, cardType);
-  merchCard.append(inner);
+  merchCard.append(bodySlot);
 };
 
-const decorateRibbon = (el, ribbonMetadata, cardType) => {
+const returnRibbonStyle = (ribbonMetadata) => {
   const ribbonStyleRegex = /^#[0-9a-fA-F]+, #[0-9a-fA-F]+$/;
-  if (!ribbonStyleRegex.test(ribbonMetadata[0]?.innerText)) return;
-  const ribbonStyle = ribbonMetadata[0].innerText;
-  const [ribbonBgColor, ribbonTextColor] = ribbonStyle.split(', ');
+  if (!ribbonStyleRegex.test(ribbonMetadata[0]?.innerText)) return null;
+  const style = ribbonMetadata[0].innerText;
+  const badgeBackgroundColor = style.split(',')[0].trim();
+  const badgeColor = style.split(',')[1].trim();
   const ribbonWrapper = ribbonMetadata[0].parentNode;
-  const ribbon = ribbonMetadata[1];
-  ribbon.classList.add(`consonant-${cardType}-ribbon`);
-  const borderStyle = `1px solid ${ribbonBgColor}`;
-  if (el.classList.contains('evergreen')) {
-    ribbon.style.border = borderStyle;
-    ribbon.style.borderRight = 'none';
-  } else {
-    ribbon.style.backgroundColor = ribbonBgColor;
-    el.style.border = borderStyle;
-  }
-  ribbon.style.color = ribbonTextColor;
-  const picture = el.querySelector(`.consonant-${cardType}-img`);
-  if (picture) {
-    picture.insertAdjacentElement('afterend', ribbon);
-  } else {
-    el.insertAdjacentElement('beforeend', ribbon);
-  }
+  const badgeText = ribbonMetadata[1].innerText;
   ribbonWrapper.remove();
+  return { badgeBackgroundColor, badgeColor, badgeText };
 };
 
-const decorateIcon = (el, icons, cardType) => {
-  if (!icons) return;
-  const inner = el.querySelector(`.consonant-${cardType}-inner`);
-  const iconWrapper = createTag('div', { class: `consonant-${cardType}-iconWrapper` });
-  icons?.forEach((icon) => {
-    const url = icon.querySelector('img').src;
-    const iconDiv = createTag('div', { class: 'consonant-MerchCard-ProductIcon', style: `background-image: url(${url})` });
-    iconWrapper.appendChild(iconDiv);
-    icon.parentNode?.remove();
-  });
-  inner.prepend(iconWrapper);
+const getActionMenuContent = (el, ribbonMetadata) => {
+  const index = (ribbonMetadata !== null) ? 1 : 0;
```

Comment:
can probably remove `()` to match line below.

Reply:
Fixed